integration-tests/src/fees.rs

```diff
@@ -3,35 +3,85 @@
 use crate::kusama_test_net::*;
 use basilisk_runtime::TransactionPayment;
 use frame_support::weights::GetDispatchInfo;
+use primitives::nft::ClassType;
 use sp_runtime::codec::Encode;
 use xcm_emulator::TestExt;
 
+macro_rules! assert_eq_approx {
+	( $x:expr, $y:expr, $z:expr, $r:expr) => {{
+		let diff = if $x >= $y { $x - $y } else { $y - $x };
+		if diff > $z {
+			panic!("\n{} not equal\nleft: {:?}\nright: {:?}\n", $r, $x, $y);
+		}
+	}};
+}
+
+//NOTE: rust encoded call size is differen from UI encoded call size that's why we have asserts for 2 fees.
+
 #[test]
-// #[ignore]
-//TODO: fix this, fees calculcated in this test are apporx 2bsx off.
 fn transfer_transaction_fees() {
-	//This test is not correct
 	Basilisk::execute_with(|| {
+		let diff = UNITS / 100; //0.01
+
+		let expected_rust_encoded_fees = 45_56 * UNITS / 100; //45.56
+		let expected_ui_fees = 4_655 * UNITS / 100; //46.55
+
 		let call = orml_currencies::Call::<basilisk_runtime::Runtime>::transfer {
 			dest: AccountId::from(ALICE),
 			currency_id: 0,
 			amount: 50 * UNITS,
 		};
 
 		let info = call.get_dispatch_info();
+		//rust encoded fees
+		let len = call.encoded_size() as u32;
+		let rust_encoded_fees = TransactionPayment::compute_fee(len, &info, 0);
+
+		assert_eq_approx!(
+			rust_encoded_fees,
+			expected_rust_encoded_fees,
+			diff,
+			"Rust encoded fees are"
+		);
+
+		//UI encoded fees
+		let len = 143; //UI encoded length
+		let ui_fees = TransactionPayment::compute_fee(len, &info, 0);
+		assert_eq_approx!(ui_fees, expected_ui_fees, diff, "UI fees are");
+
+		// Min fee adjustment multiplier
+		pallet_transaction_payment::pallet::NextFeeMultiplier::<basilisk_runtime::Runtime>::put(
+			basilisk_runtime::MinimumMultiplier::get(),
+		);
 		let len = call.encoded_size() as u32;
-		let fees = TransactionPayment::compute_fee(len, &info, 0);
+		let min_multiplier_rust_fees = TransactionPayment::compute_fee(len, &info, 0);
 
-		//This test is not correct it's approx 1bsx of from real fees
-		println!("transfer cost {:?} BSX", fees / UNITS);
+		let len = 143; //UI encoded length
+		let min_multiplier_ui_fees = TransactionPayment::compute_fee(len, &info, 0);
+
+		println!(
```

Comment:
why do we print all the fee values at the end of these tests? Do we need verify it visually?

Reply:
and what is the reason for that? 

I mean, why it is better than using some assertions in the code?

Reply:
There are assertions in the code but they are not precise. It's up to us said if the diff is ok. 
Also when you are changing fees it's easier to see them, than running whole chain and look into UI.
This things are not printed by default so I don't see any downside of having them

Reply:
@martinfridrich fair enough, sound reasonable then!